### PR TITLE
Fix clippy and rustdoc warnings

### PR DESCRIPTION
**Pull request by Connortsui20**
Fixes a few rustdoc and clippy warnings.

The only thing that might not be good is my removal of the `try_from()` and `unwrap()`, but since it won't type check if the sys type changes I think it's fine.

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -393,13 +393,14 @@ impl<'buf> RecvMsgOut<'buf> {
...
     #[allow(clippy::result_unit_err)]
+    #[allow(clippy::useless_conversion)]
     pub fn parse(buffer: &'buf [u8], msghdr: &libc::msghdr) -> Result<Self, ()> {
         let msghdr_name_len = usize::try_from(msghdr.msg_namelen).unwrap();
         let msghdr_control_len = usize::try_from(msghdr.msg_controllen).unwrap();
 
         if Self::DATA_START
             .checked_add(msghdr_name_len)
-            .and_then(|acc| acc.checked_add(msghdr_control_len))
+            .and_then(|acc| acc.checked_add(usize::try_from(msghdr.msg_controllen).unwrap()))
```
Comment (quininer) [inline src/types.rs:403]: this seems unnecessary

Comment (Connortsui20) [reply src/types.rs:403]: It's only to pass the CI

Comment (Connortsui20) [reply src/types.rs]: Whoops forgot to change that back

**Comment by quininer** [comment]
Thank you!

**Pull request merged**